pkg/app/var.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/resource"
+	"github.com/kubernetes-sigs/kustomize/pkg/types"
+)
+
+func gvkn(rv types.Var) resource.ResId {
```

Comment:
Please make this a method on the types.Var class. 

Because the rest of the code here is moving out, and it would be odd just keeping this snippet in a file called var.go when there's a var class elsewhere.

I don't think it creates an import cycle, but if it does then just inline the code at the call point.

Reply:
I removed this function by directly calling `resource.NewResId`